### PR TITLE
New pull request including DMA fifo as well.

### PR DESCRIPTION
**Pull request by lam-borg**
new drivers for NXP RT1176

- modified Kconfigs to make the new drivers selectable
- added advanced PWM driver
- added dma fifo

### CODE REVIEW

File: drivers/dma/dma_mcux_fifo.c [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	cache_data_enable();
+	return true;
+}
+
+static int dma_mcux_fifo_init(const struct device *dev)
+{
+	cache_data_disable();
+	dma_mcux_fifo_data_t *data = (dma_mcux_fifo_data_t *)dev->data;
+	dma_mcux_fifo_config_t *config = (dma_mcux_fifo_config_t *)dev->config;
+	data->fifo_data = k_malloc(pow(2, config->fifo_size_log2));
```
Comment (mas-sieve) [inline drivers/dma/dma_mcux_fifo.c:63]: I think `pow` returns a float or double. I suggest to either cast the result explicitly back to `size_t` or just use a bit shift.

 ```suggestion
	data->fifo_data = k_malloc((size_t)pow(2, config->fifo_size_log2));
```

```suggestion
	data->fifo_data = k_malloc( 1u << fifo_size_log2);
```

File: drivers/dma/dma_mcux_fifo.c [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+	/*if a flag needs to be set/reset after reading, the minor loop linked channel needs to be implemented*/
+	edma_tcd_t *slave = (edma_tcd_t *)&edma->TCD[config->dma_channel2];
+	edma_transfer_config_t master_config = {.srcAddr = config->source_address,
+					       .destAddr = (uint32_t)data->fifo_data,
+					       .srcTransferSize = 1,
+					       .destTransferSize = 1,
+					       .srcOffset = 0,
+					       .destOffset = config->source_width,
+					       .minorLoopBytes = config->source_width,
+					       .majorLoopCounts = pow(2, config->fifo_size_log2) /
```
Comment (mas-sieve) [inline drivers/dma/dma_mcux_fifo.c:75]:
```suggestion
					       .majorLoopCounts = ((size_t)pow(2, config->fifo_size_log2)) /
```

```suggestion
					       .majorLoopCounts = (1u << fifo_size_log2) /
```

File: dts/bindings/dma/nxp,mcux-fifo.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# Copyright (c) 2019, Linaro Limited
```
Comment (mas-sieve) [inline dts/bindings/dma/nxp,mcux-fifo.yaml:1]: If you add new files, keeping the copyright notice of another company is probably a bad idea.

File: include/zephyr/drivers/dma/dma_mcux_edma.h [hunk 4]
```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2020 NXP Semiconductor INC.
```
Comment (mas-sieve) [inline include/zephyr/drivers/dma/dma_mcux_edma.h:2]: Same here.

Comment (lam-borg) [reply include/zephyr/drivers/dma/dma_mcux_edma.h:2]: This is really not my work... i just edited something.
Dont know how to handle that...

File: drivers/dma/dma_mcux_fifo.c (hunk 2 above)
Comment (lam-borg) [reply drivers/dma/dma_mcux_fifo.c:75]: jup, makes sense

File: drivers/dma/dma_mcux_fifo.c (hunk 1 above)
Comment (lam-borg) [reply drivers/dma/dma_mcux_fifo.c:63]: jup, makes sense

File: dts/bindings/dma/nxp,mcux-fifo.yaml (hunk 3 above)
Comment (lam-borg) [reply dts/bindings/dma/nxp,mcux-fifo.yaml:1]: fixed. also added description

**Review by mas-sieve — APPROVED** [review]

**Pull request merged**